### DISCUSSION TITLE
Unable to jump to definition

### DISCUSSION THREAD

**Discussion by shiena** [godot-zig/godot-zig#15] [General]
I cannot jump from `const GD = @import("gen").GodotCore;` to GodotCore.zig in src/api/Godot.zig with the following combination.
Can this be resolved by modifying build.zig? Or is it a problem on the zls side?

- vscode
- zls: 0.12.0
- zig: 0.12.0
- godot-zig: 3ac053156cd016ae27af9d46c7ebec67f3bc0043

**Comment by elvisxzhou** [comment]
Clicking on "GodotCore" triggers a jump to GodotCore.zig in my case.
- vscode 1.88.1
- zls 0.12.0-dev.522+f03d545
- zig 0.12.1-dev.17+600b65282
- godot-zig master

**Reply by shiena** [reply]
Hmm, in my environment, Variant included in the current folder is recognized, but GodotCore and UtilityFunctions included in the gen namespace are not recognized, as shown below.
GodotCore.zig is generated in the `zig-cache/o/<hash>` folder because it is after executing zig build.
What could be the possible cause?

![image](https://github.com/godot-zig/godot-zig/assets/1295639/ba6e1483-db16-4339-9991-9e79a6e8cbdd)
![image](https://github.com/godot-zig/godot-zig/assets/1295639/74beb7fe-8ef9-4f0e-970a-4fa36909d588)

**Reply by shiena** [reply]
The issue has not been resolved but I have found similar challenges.
https://github.com/zigtools/zls/issues/1851

**Comment by silversquirl** [comment]
Could you send the ZLS log? Open the output pane (command panel -> `Output: Focus on Output View`) then choose "Zig Language Server" from the selector in the top right.

**Reply by silversquirl** [reply]
Would also be good to know if other ZLS features work with godot-zig for you, like completions or documentation popups, though I expect the answer is that they don't.

**Reply by shiena** [reply]
This is the log of zls when opening src/api/Godot.zig from the godot-zig folder in vscode.
[zls.log](https://github.com/godot-zig/godot-zig/files/15241589/zls.log)

**Reply by silversquirl** [reply]
You're using the wrong version of ZLS - it's using the copy at `C:\Users\shiena\.local\bin\zls.exe`, which is version `0.13.0-dev.24+a660910`.
If you have ZLS 0.12 installed somewhere else, you'll need to configure VSCode with the correct path, it's the `Zig > ZLS: Path` option in user settings.

**Reply by shiena** [reply]
Re-logged with zls 0.12.0. Definition jumps are not available over here either.
[zls.1.log](https://github.com/godot-zig/godot-zig/files/15244136/zls.1.log)

**Reply by silversquirl** [reply]
Just to be completely sure: have you run `zig build` at least once in order to generate the bindings?
If so (or if that still doesn't fix it), could you clear the output log, attempt a single jump-to-def on `GodotCore`, then send just the log that that generates? Just makes it a bit easier to narrow down :)

**Reply by shiena** [reply]
Yes, it is generated in the `zig-cache/o` folder as shown in the screenshot by running `zig build`.

![image](https://github.com/godot-zig/godot-zig/assets/1295639/b8c22b93-2ae4-47dd-aff9-4eec02626c6f)
